scripts/ci-entrypoint.sh

```diff
@@ -152,6 +155,17 @@ create_cluster
 # export the target cluster KUBECONFIG if not already set
 export KUBECONFIG="${KUBECONFIG:-${PWD}/kubeconfig}"
 
+# install cloud-provider-azure components, if using out-of-tree
+if [[ -n "${TEST_CCM:-}" ]]; then
```

Comment:
Validated this new ccm + helm scenario:

```sh
s$ TEST_CCM=true AZURE_LOADBALANCER_SKU=standard CONTROL_PLANE_MACHINE_COUNT=3 REGISTRY=docker.io/jackfrancis KUBERNETES_VERSION=1.23.5 ./scripts/ci-entrypoint.sh
...
# Get kubeconfig and store it locally.
kubectl get secrets capz-q4tba7-kubeconfig -o json | jq -r .data.value | base64 --decode > ./kubeconfig
timeout --foreground 600 bash -c "while ! kubectl --kubeconfig=./kubeconfig get nodes | grep control-plane; do sleep 1; done"
error: the server doesn't have a resource type "nodes"
capz-q4tba7-control-plane-s95kf   NotReady   control-plane,master   6s    v1.23.5
run "kubectl --kubeconfig=./kubeconfig ..." to work with the new target cluster
WARNING: Kubernetes configuration file is group-readable. This is insecure. Location: /Users/jackfrancis/go/src/sigs.k8s.io/cluster-api-provider-azure/kubeconfig
WARNING: Kubernetes configuration file is world-readable. This is insecure. Location: /Users/jackfrancis/go/src/sigs.k8s.io/cluster-api-provider-azure/kubeconfig
Installing cloud-provider-azure components via helm
NAME: cloud-provider-azure-1649354900
LAST DEPLOYED: Thu Apr  7 11:08:22 2022
NAMESPACE: default
STATUS: deployed
REVISION: 1
TEST SUITE: None
```

Manually validated that things looked as expected:

```sh
$ k get pods -A
NAMESPACE     NAME                                                      READY   STATUS    RESTARTS        AGE
kube-system   calico-kube-controllers-7c664f4648-sth6w                  1/1     Running   0               6m49s
kube-system   calico-node-79nft                                         1/1     Running   0               6m49s
kube-system   calico-node-mjvl4                                         1/1     Running   0               5m39s
kube-system   calico-node-sn5mg                                         1/1     Running   0               5m42s
kube-system   calico-node-svsnb                                         1/1     Running   0               3m9s
kube-system   calico-node-v5dhf                                         1/1     Running   0               5m9s
kube-system   cloud-controller-manager-69fcc57c84-b69pq                 1/1     Running   1 (4m57s ago)   6m43s
kube-system   cloud-node-manager-4n7cf                                  1/1     Running   0               6m43s
kube-system   cloud-node-manager-565r4                                  1/1     Running   0               5m42s
kube-system   cloud-node-manager-ck497                                  1/1     Running   0               3m9s
kube-system   cloud-node-manager-qr9zw                                  1/1     Running   0               5m8s
kube-system   cloud-node-manager-qvhlw                                  1/1     Running   0               5m39s
kube-system   coredns-64897985d-fhhqj                                   1/1     Running   0               6m50s
kube-system   coredns-64897985d-gtnjp                                   1/1     Running   0               6m50s
kube-system   etcd-capz-q4tba7-control-plane-7wss9                      1/1     Running   0               3m7s
kube-system   etcd-capz-q4tba7-control-plane-gpplv                      1/1     Running   0               5m7s
kube-system   etcd-capz-q4tba7-control-plane-s95kf                      1/1     Running   0               6m45s
kube-system   kube-apiserver-capz-q4tba7-control-plane-7wss9            1/1     Running   0               3m8s
kube-system   kube-apiserver-capz-q4tba7-control-plane-gpplv            1/1     Running   0               5m8s
kube-system   kube-apiserver-capz-q4tba7-control-plane-s95kf            1/1     Running   0               6m54s
kube-system   kube-controller-manager-capz-q4tba7-control-plane-7wss9   1/1     Running   0               3m8s
kube-system   kube-controller-manager-capz-q4tba7-control-plane-gpplv   1/1     Running   0               5m8s
kube-system   kube-controller-manager-capz-q4tba7-control-plane-s95kf   1/1     Running   1 (4m58s ago)   6m54s
kube-system   kube-proxy-74nrq                                          1/1     Running   0               5m9s
kube-system   kube-proxy-79bj8                                          1/1     Running   0               6m50s
kube-system   kube-proxy-88dl6                                          1/1     Running   0               3m9s
kube-system   kube-proxy-brhzh                                          1/1     Running   0               5m39s
kube-system   kube-proxy-x72bc                                          1/1     Running   0               5m42s
kube-system   kube-scheduler-capz-q4tba7-control-plane-7wss9            1/1     Running   0               3m8s
kube-system   kube-scheduler-capz-q4tba7-control-plane-gpplv            1/1     Running   0               5m7s
kube-system   kube-scheduler-capz-q4tba7-control-plane-s95kf            1/1     Running   1 (4m57s ago)   6m45s
$ k get pod cloud-controller-manager-69fcc57c84-b69pq -n kube-system -o yaml | grep 'image:'
    image: docker.io/jackfrancis/azure-cloud-controller-manager:4f3bab6
    image: docker.io/jackfrancis/azure-cloud-controller-manager:4f3bab6
k get pod cloud-node-manager-4n7cf -n kube-system -o yaml | grep 'image:'
    image: docker.io/jackfrancis/azure-cloud-node-manager:4f3bab6
    image: docker.io/jackfrancis/azure-cloud-node-manager:4f3bab6
```